core/test/matrix/batch_csr.cpp

```diff
@@ -0,0 +1,549 @@
+        EXPECT_EQ(m->get_const_values()[1], value_type{2.5});
+        EXPECT_EQ(m->get_const_values()[2], value_type{3.5});
+        EXPECT_EQ(m->get_const_values()[3], value_type{1.0});
+        EXPECT_EQ(m->get_const_values()[4], value_type{2.0});
+        EXPECT_EQ(m->get_const_values()[5], value_type{3.0});
+        EXPECT_EQ(m->get_const_col_idxs()[0], index_type{0});
+        EXPECT_EQ(m->get_const_col_idxs()[1], index_type{1});
+        EXPECT_EQ(m->get_const_col_idxs()[2], index_type{2});
+        EXPECT_EQ(m->get_const_row_ptrs()[0], index_type{0});
+        EXPECT_EQ(m->get_const_row_ptrs()[1], index_type{1});
+        EXPECT_EQ(m->get_const_row_ptrs()[2], index_type{3});
+    }
+
+    static void assert_equal_to_original_mtx(const BatchCsrMtx* m)
+    {
+        ASSERT_EQ(m->get_num_batch_items(), 2);
+        ASSERT_EQ(m->get_common_size(), gko::dim<2>(2, 3));
+        ASSERT_EQ(m->get_num_stored_elements(), 2 * 6);
+        EXPECT_EQ(m->get_const_values()[0], value_type{-1.0});
+        EXPECT_EQ(m->get_const_values()[1], value_type{2.0});
+        EXPECT_EQ(m->get_const_values()[2], value_type{3.0});
+        EXPECT_EQ(m->get_const_values()[3], value_type{-1.5});
+        EXPECT_EQ(m->get_const_values()[4], value_type{2.5});
+        EXPECT_EQ(m->get_const_values()[5], value_type{3.5});
+        EXPECT_EQ(m->get_const_values()[6], value_type{1.0});
+        EXPECT_EQ(m->get_const_values()[7], value_type{2.5});
+        EXPECT_EQ(m->get_const_values()[8], value_type{3.0});
+        EXPECT_EQ(m->get_const_values()[9], value_type{1.0});
+        EXPECT_EQ(m->get_const_values()[10], value_type{2.0});
+        EXPECT_EQ(m->get_const_values()[11], value_type{3.0});
+        EXPECT_EQ(m->get_const_col_idxs()[0], index_type{0});
+        EXPECT_EQ(m->get_const_col_idxs()[1], index_type{1});
+        EXPECT_EQ(m->get_const_col_idxs()[2], index_type{2});
+        EXPECT_EQ(m->get_const_col_idxs()[3], index_type{0});
+        EXPECT_EQ(m->get_const_col_idxs()[4], index_type{1});
+        EXPECT_EQ(m->get_const_col_idxs()[5], index_type{2});
+        EXPECT_EQ(m->get_const_row_ptrs()[0], index_type{0});
+        EXPECT_EQ(m->get_const_row_ptrs()[1], index_type{3});
+        EXPECT_EQ(m->get_const_row_ptrs()[2], index_type{6});
+    }
+
+    static void assert_empty(BatchCsrMtx* m)
+    {
+        ASSERT_EQ(m->get_num_batch_items(), 0);
+        ASSERT_EQ(m->get_num_stored_elements(), 0);
+    }
+
+    std::shared_ptr<const gko::Executor> exec;
+    std::unique_ptr<BatchCsrMtx> mtx;
+    std::unique_ptr<BatchCsrMtx> sp_mtx;
+    std::unique_ptr<CsrMtx> csr_mtx;
+    std::unique_ptr<CsrMtx> sp_csr_mtx;
+};
+
+TYPED_TEST_SUITE(Csr, gko::test::ValueTypes, TypenameNameGenerator);
+
+
+TYPED_TEST(Csr, KnowsItsSizeAndValues)
+{
+    this->assert_equal_to_original_mtx(this->mtx.get());
+}
+
+
+TYPED_TEST(Csr, SparseMtxKnowsItsSizeAndValues)
+{
+    this->assert_equal_to_original_sparse_mtx(this->sp_mtx.get());
+}
+
+
+// TYPED_TEST(Csr, CanBeEmpty)
```

Comment:
I don't think we should commit commented out code. you can disable the tests by prefixing them with `DISABLED_` IIRC